src/ChronosInterval.php

```diff
@@ -166,298 +74,60 @@ public function __construct(
+        $spec = static::PERIOD_PREFIX;
 
-            case 'microseconds':
-            case 'microsecond':
-                return new static(null, null, null, null, null, null, null, $arg);
+        if ($years) {
+            $spec .= $years . static::PERIOD_YEARS;
         }
-
-        throw new BadMethodCallException(sprintf('Cannot create interval with `%s` units', $name));
-    }
-
-    /**
-     * Create a ChronosInterval instance from a DateInterval one.  Can not instance
-     * DateInterval objects created from DateTime::diff() as you can't externally
-     * set the $days field.
-     *
-     * @param \DateInterval $interval The DateInterval instance to copy.
-     * @throws \InvalidArgumentException
-     * @return static
-     */
-    public static function instance(DateInterval $interval): static
-    {
-        if (static::wasCreatedFromDiff($interval)) {
-            throw new InvalidArgumentException(
-                'Can not instance a DateInterval object created from DateTime::diff().'
-            );
+        if ($months) {
+            $spec .= $months . static::PERIOD_MONTHS;
         }
-
-        $instance = new static($interval->y, $interval->m, 0, $interval->d, $interval->h, $interval->i, $interval->s);
-        $instance->f = $interval->f;
-        $instance->invert = $interval->invert;
-        $instance->days = $interval->days;
-
-        return $instance;
-    }
-
-    /**
-     * Get a part of the ChronosInterval object
-     *
-     * @param string $name The property to read.
-     * @throws \InvalidArgumentException
-     * @return int
-     */
-    public function __get(string $name): int
-    {
-        switch ($name) {
-            case 'years':
-                return $this->y;
-
-            case 'months':
-                return $this->m;
-
-            case 'dayz':
-                return $this->d;
-
-            case 'hours':
-                return $this->h;
-
-            case 'minutes':
-                return $this->i;
-
-            case 'seconds':
-                return $this->s;
-
-            case 'microseconds':
-                return (int)($this->f * 1000000);
-
-            case 'weeks':
-                return (int)floor($this->d / ChronosInterface::DAYS_PER_WEEK);
-
-            case 'daysExcludeWeeks':
-            case 'dayzExcludeWeeks':
-                return $this->dayz % ChronosInterface::DAYS_PER_WEEK;
-
-            default:
-                throw new InvalidArgumentException(sprintf("Unknown getter '%s'", $name));
+        if ($weeks) {
+            $spec .= $weeks . static::PERIOD_WEEKS;
         }
-    }
-
-    /**
-     * Set a part of the ChronosInterval object
-     *
-     * @param string $name The property to augment.
-     * @param int|false $val The value to change.
-     * @return void
-     * @throws \InvalidArgumentException
-     */
-    public function __set(string $name, int|false $val): void
```

Comment:
Without these mutation methods how would users modify or adjust an interval once it has been made? Should we also have a `modifyWith()` method that has a similar interface to `create()` that returns a new updated interval?

Reply:
Are you aware of the use cases for needing to modify an interval? `DateInterval` doesn't  provided any APi to do so. The only way to modify an interval after creation is manipulating it properties directly and the PHP manuals warns against doing so and treating the methods as read only.

Me and @othercorey discussed this a bit more and think just having a `Chronos::createInternal()` which reruns a `DateInterval` would suffice and we can drop `ChronosInterval` all together.

Reply:
I don't have use cases in mind for modifying an interval. We can always add more API surface in the future :smile: